### PR TITLE
Release/alpha 0.0.1

### PR DESCRIPTION
**Pull request by MariyaSin**

### CODE REVIEW

**Review by djino97 — APPROVED** [review]

File: src/ProjectService.Broker/DisactivateUserConsumer.cs [hunk 1]
```diff
@@ -17,7 +17,7 @@ public class DisactivateUserConsumer : IConsumer<IDisactivateUserRequest>
 
         public Task Consume(ConsumeContext<IDisactivateUserRequest> context)
         {
-            _repository.Remove(context.Message.UserId);
+            _repository.Remove(context.Message.UserId, context.Message.ModifiedBy);
```
Comment (Glunky) [inline src/ProjectService.Broker/DisactivateUserConsumer.cs:20]: may move context.Message to a variable. but its not necessary

File: src/ProjectService.Broker/GetProjectUsersConsumer.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+namespace LT.DigitalOffice.ProjectService.Broker
+{
+    public class GetProjectUsersConsumer : IConsumer<IGetProjectUsersRequest>
+    {
+        private readonly IUserRepository _userRepository;
+
+        private object GetProjectUsers(IGetProjectUsersRequest request)
+        {
+            var users = _userRepository.Find(request.ProjectUsers.Select(pu => pu.userId).ToList());
```
Comment (Glunky) [inline src/ProjectService.Broker/GetProjectUsersConsumer.cs:18]: remove var

Comment (Glunky) [reply src/ProjectService.Broker/GetProjectUsersConsumer.cs:18]: may ProjectUsers move to a variable

File: src/ProjectService.Business/Commands/Project/CreateProjectCommand.cs [hunk 3]
```diff
@@ -34,36 +31,75 @@ public class CreateProjectCommand : ICreateProjectCommand
...
-        private IGetDepartmentResponse GetDepartment(Guid departmentId, List<string> errors)
+        private List<Guid> CheckDepartmentExistence(Guid? departmentId, List<string> errors)
         {
-            string errorMessage = "Cannot add project. Please try again later.";
+            if (!departmentId.HasValue)
+            {
+                return null;
+            }
+
+            string errorMessage = "Failed to check the existing department.";
```
Comment (Glunky) [inline src/ProjectService.Business/Commands/Project/CreateProjectCommand.cs:47]: errorMessage nowhere used except before return null. Do not set errorMsg to a variable - set it to args

File: src/ProjectService.Business/Commands/Project/CreateProjectCommand.cs [hunk 4]
```diff
@@ -34,36 +31,75 @@ public class CreateProjectCommand : ICreateProjectCommand
...
+            string logMessage = "Department with id: {id} not found.";
 
             try
             {
-                var response = _rcGetDepartment.GetResponse<IOperationResult<IGetDepartmentResponse>>(
-                    IGetDepartmentRequest.CreateObj(null, departmentId), timeout: TimeSpan.FromSeconds(2)).Result;
-
+                var response = _rcCheckDepartmentsExistence.GetResponse<IOperationResult<ICheckDepartmentsExistence>>(
```
Comment (Glunky) [inline src/ProjectService.Business/Commands/Project/CreateProjectCommand.cs:52]: do not use var here

File: src/ProjectService.Business/Commands/Project/CreateProjectCommand.cs [hunk 5]
```diff
@@ -34,36 +31,75 @@ public class CreateProjectCommand : ICreateProjectCommand
...
+                    ICheckDepartmentsExistence.CreateObj(new List<Guid> { departmentId.Value })).Result;
                 if (response.Message.IsSuccess)
                 {
-                    return response.Message.Body;
+                    if (!response.Message.Body.DepartmentIds.Any())
```
Comment (Glunky) [inline src/ProjectService.Business/Commands/Project/CreateProjectCommand.cs:56]: response.Message.Body to a variable

File: src/ProjectService.Business/Commands/Project/CreateProjectCommand.cs [hunk 6]
```diff
@@ -34,36 +31,75 @@ public class CreateProjectCommand : ICreateProjectCommand
...
+        }
+
+        private List<Guid> CheckUserExistence(List<Guid> userIds, List<string> errors)
+        {
+            if (!userIds.Any())
+            {
+                return userIds;
             }
 
+            string errorMessage = "Failed to check the existing users.";
```
Comment (Glunky) [inline src/ProjectService.Business/Commands/Project/CreateProjectCommand.cs:82]: errorMessage nowhere used except in catch. Do not set errorMsg to a variable - set it to args. logMessage also

File: src/ProjectService.Business/Commands/Project/CreateProjectCommand.cs [hunk 7]
```diff
@@ -162,34 +199,27 @@ public OperationResultResponse<ProjectInfo> Execute(ProjectRequest request)
 
             _validator.ValidateAndThrowCustom(request);
 
-            IGetDepartmentResponse department = null;
-            if (request.DepartmentId.HasValue)
+            var existUsers = CheckUserExistence(request.Users.Select(u => u.UserId).ToList(), response.Errors);
```
Comment (Glunky) [inline src/ProjectService.Business/Commands/Project/CreateProjectCommand.cs:202]: do not use var here

File: src/ProjectService.Business/Commands/Project/CreateProjectCommand.cs [hunk 8]
```diff
@@ -34,36 +31,75 @@ public class CreateProjectCommand : ICreateProjectCommand
...
+            {
+                return userIds;
             }
 
+            string errorMessage = "Failed to check the existing users.";
+            string logMessage = "Cannot check existing users withs this ids {userIds}";
+
+            try
+            {
+                var response = _rcCheckUsersExistence.GetResponse<IOperationResult<ICheckUsersExistence>>(
```
Comment (Glunky) [inline src/ProjectService.Business/Commands/Project/CreateProjectCommand.cs:87]: do not use var

File: src/ProjectService.Business/Commands/Project/CreateProjectCommand.cs [hunk 9]
```diff
@@ -162,34 +199,27 @@ public OperationResultResponse<ProjectInfo> Execute(ProjectRequest request)
 
             _validator.ValidateAndThrowCustom(request);
 
-            IGetDepartmentResponse department = null;
-            if (request.DepartmentId.HasValue)
+            var existUsers = CheckUserExistence(request.Users.Select(u => u.UserId).ToList(), response.Errors);
+            if (!response.Errors.Any()
```
Comment (Glunky) [inline src/ProjectService.Business/Commands/Project/CreateProjectCommand.cs:203]: response.errors to a variable

**Review by Glunky — APPROVED** [review]

**Pull request merged**